### ISSUE TITLE
Duplicate Callback Triggers in RabbitMQ Integration

### ISSUE THREAD

**Issue by sinhaugoh** [nonblocking/mashroom#115]
Hi there,

## Issue:
In certain scenarios, the subscriber callback seems to trigger twice, even when there's only one message incoming.

## Steps to reproduce
1. Create a remote plugin that utilises `messageBus` to subscribe to 2 remote topics:
    - notifications/#
    - notifications/case/#

```sm
    const topic1 = `${messageBus.getRemotePrefix()}notifications/#`;
    messageBus.subscribe(topic1, (data) => {
      console.log(`received message in topic: ${topic1} `);
    });

    const topic2 = `${messageBus.getRemotePrefix()}notifications/case/#`;
    messageBus.subscribe(topic2, (data) => {
      console.log(`received message in topic: ${topic2} `);
    });

```

2. Send a message with this routing key: "mashroom.notifications.case.abc" 

## Logs
```
// Initial state
Sending WebSocket message: {"messageId":"9HNFmlxr","command":"subscribe","topic":"notifications/#"}
Sending WebSocket message: {"messageId":"V19JurSF","command":"subscribe","topic":"notifications/case/#"}
Received WebSocket message: {"messageId":"9HNFmlxr","success":true}
Received WebSocket message: {"messageId":"V19JurSF","success":true}

// Received a rabbitMQ message (the callback triggered twice)
Received WebSocket message: {"remoteMessage":true,"topic":"notifications/case/abc","message":{"id":1,"name":"Alan Turing"}}
Received remote message for topic:  remote:notifications/case/abc
received message in topic: remote:notifications/# 
received message in topic: remote:notifications/case/# 
Received WebSocket message: {"remoteMessage":true,"topic":"notifications/case/abc","message":{"id":1,"name":"Alan Turing"}}
Received remote message for topic:  remote:notifications/case/abc
received message in topic: remote:notifications/# 
received message in topic: remote:notifications/case/# 
```

**Comment by nonblocking** [comment]
The "#" is a multi-level wildcard, so it behaves exactly as expected, because both "notifications/#" and "notifications/case/#" match "notifications/case/abc".

Not sure what you would have expected, but maybe it helps to replace "#" by "+" (single level wildcard).

**Comment by sinhaugoh** [comment]
Is it normal that both consumers of "notifications/#" and "notification/case/#" receive the same message twice when there is only one message published to "notifications/case/abc"?

**Comment by nonblocking** [comment]
Yes, because both subscriptions match the notification path. What would you expect?

**Comment by sinhaugoh** [comment]
I expect both subscriptions to match the notification path, but each of them triggers the callback once.

So, from the example, I should receive this:

received message in topic: remote:notifications/# 
received message in topic: remote:notifications/case/# 

Not:

received message in topic: remote:notifications/# 
received message in topic: remote:notifications/case/# 
received message in topic: remote:notifications/# 
received message in topic: remote:notifications/case/#

**Comment by nonblocking** [comment]
Ok, I get it. That happens because the client-side message bus doesn't know at the moment for which subscription a message exactly is and just checks which topic patterns would match.

We are going to fix this.

**Comment by nonblocking** [comment]
Fixed in master and the upcoming version 2.5.5
